app/page.tsx

```diff
@@ -1,43 +1,99 @@
 import { types } from "@mozilla/nimbus-shared";
-import { BranchInfo, ExperimentAndBranchInfo, ExperimentInfo, experimentColumns, FxMSMessageInfo, fxmsMessageColumns } from "./columns";
-import { getDashboard, getDisplayNameForTemplate, getTemplateFromMessage, _isAboutWelcomeTemplate } from "../lib/messageUtils.ts";
+import {
+  BranchInfo,
+  ExperimentAndBranchInfo,
+  ExperimentInfo,
+  experimentColumns,
+  FxMSMessageInfo,
+  fxmsMessageColumns,
+} from "./columns";
+import {
+  getDashboard,
+  getDisplayNameForTemplate,
+  getTemplateFromMessage,
+  _isAboutWelcomeTemplate,
+} from "../lib/messageUtils.ts";
 
 import { MessageTable } from "./message-table";
-import { getProposedEndDate, usesMessagingFeatures } from "../lib/experimentUtils.ts";
+import {
+  getProposedEndDate,
+  usesMessagingFeatures,
+} from "../lib/experimentUtils.ts";
 import Link from "next/link";
 
-function getASRouterLocalColumnFromJSON(messageDef: any) : FxMSMessageInfo {
-  let fxmsMsgInfo : FxMSMessageInfo = {
-    product: 'Desktop',
-    release: 'Fx 123',
+function _substituteLocalizations(values, localizations) {
```

Comment:
Let's put this into experimentUtils.ts and add tests for at least important stuff.

Reply:
Also, adding types for this function would be worthwhile, and might solve some of the complaints from the compiler in `npm run build`.